test/test_constrained_dataset.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+import numpy as np
+from metric_learn.constraints import ConstrainedDataset
+from numpy.testing import assert_array_equal
+from sklearn.model_selection import StratifiedKFold, KFold
+from sklearn.utils.testing import assert_raise_message
+
+X = np.random.randn(20, 5)
+c = np.random.randint(0, X.shape[0], (15, 2))
+cd = ConstrainedDataset(X, c)
+y = np.random.randint(0, 2, c.shape[0])
+group = np.random.randint(0, 3, c.shape[0])
+
+c_shape = c.shape[0]
```

Comment:
This is only used by `test_getitem`, so I'd move it there. (And probably name it `num_constraints`)

Reply:
actually this is also used in `test_shape` (although not through this variable)

Reply:
I agree I will rename it and put it before the definitions that use it